dataset/src/main/scala/frameless/Job.scala

```diff
@@ -0,0 +1,31 @@
+package frameless
+
+import org.apache.spark.SparkContext
+
+sealed abstract class Job[T](implicit sc: SparkContext) { self =>
+  /** Runs a new Spark job. */
+  def run: T
+
+  def withGroupId(groupId: String): Job[T] = {
+    withLocalProperty("spark.jobGroup.id", groupId)
+  }
+
+  def withDescription(groupId: String): Job[T] = {
+    withLocalProperty("spark.job.description", groupId)
+  }
+
+  def withLocalProperty(key: String, value: String): Job[T] = {
+    new Job[T] {
+      def run: T = {
+        sc.setLocalProperty(key, value)
+        self.run
+      }
+    }
+  }
+}
+
+object Job {
+  def apply[A](a: => A)(implicit sc: SparkContext): Job[A] = new Job[A] {
```

Comment:
Would prefer `apply` to take `SparkContext => A` and rename what you have here to be `const` or `pure` or something